op-program/client/interop/consolidate.go

```diff
@@ -273,7 +262,10 @@ func newConsolidateCheckDeps(
 		progress := transitionState.PendingProgress[i]
 		// This is the optimistic head. It's OK if it's replaced by a deposits-only block.
 		// Because by then the replacement block won't be used for hazard checks.
-		head := oracle.BlockByHash(progress.BlockHash, chain.ChainID)
+		head, err := fetchOptimisticBlock(oracle, progress.BlockHash, chain)
```

Comment:
for extra safety, could we preload the optimistic block at the beginning of the consolidation step or `RunConsolidation`? So we're future proof against block fetches that are later added prior to `newConsolidateCheckDeps`.

Reply:
Yeah I like you're thinking but I don't see a way to fit it in nicely. We'd just be adding a loop that loads the optimistic heads and throws away the result. I can do that if you want, but given we have the action tests now keeping us honest and there being very little prior to where it call newConsolidateCheckDeps I'd lean towards leaving it as is.

What do you think?

Reply:
That makes sense. It's OK to leave it as-is given the action test coverage.